### PR TITLE
feat: send emails

### PR DESCRIPTION
**Pull request by riensen**
### Background
Currently, AutoGPT is unable to send emails. This features provides to capability for AutoGPT to send simple emails using standard SMTP.

### Changes
- implemented email command: [autogpt/commands/email.py](https://github.com/Significant-Gravitas/Auto-GPT/compare/master...riensen:Auto-GPT:master#diff-a5c03ef111558c544842535f9fc24da20c82b3cbd1b40b9c4f2759743ff883aa)
- integration via [autogpt/prompt.py](https://github.com/Significant-Gravitas/Auto-GPT/compare/master...riensen:Auto-GPT:master#diff-7121a662c4bcd0ed103b6a6c8d2e5c64f1ae69cb92dd67cef4a49e2216635231) and autogpt/prompt.py https://github.com/Significant-Gravitas/Auto-GPT/compare/master...riensen:Auto-GPT:master#diff-ea57237ee734eb84f1972d815e32e279b796d84c212fde8e3cff71060a50a7b3
- updated README provides setup instructions

### Documentation
- send_email function is documented
- updated README provides setup instructions

### Test Plan
Run autogpt with goals of ['send an email to send.email.autogpt@trash-mail.com to introduce yourself'], name "CommunicatorGPT" role of "communicate". Check trash-mail.com with temporary account "send.email.autogpt" for message of style:

```
Hello,

My name is CommunicatorGPT and I am an LLM. I am writing to introduce myself and to let you know that I will be terminating shortly. Thank you for your time.

Best regards,
CommunicatorGPT
```

### PR Quality Checklist
- [X] My pull request is atomic and focuses on a single change.
- [X] I have thoroughly tested my changes with multiple different prompts.
- [X] I have considered potential risks and mitigations for my changes.
- [X] I have documented my changes clearly and comprehensively.
- [X] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

File: autogpt/commands/email.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    sender_pwd = os.environ.get("EMAIL_PASSWORD")
+
+    msg = EmailMessage()
+    msg['Subject'] = subject
+    msg['From'] = sender
+    msg['To'] = recipient
+    msg.set_content(message)
+
+# send email
+    with smtplib.SMTP('smtp.gmail.com', 587) as smtp:
```
Comment (hdkiller) [inline autogpt/commands/email.py]: Why host and port is hardcoded?

**Comment by nponeccop** [comment]
@riensen CI failed

File: .env.template [hunk 2]
```diff
@@ -21,7 +21,7 @@ AI_SETTINGS_FILE=ai_settings.yaml
 
 ### OPENAI
 # OPENAI_API_KEY - OpenAI API Key (Example: my-openai-api-key)
-# TEMPERATURE - Sets temperature in OpenAI (Default: 0)
+# TEMPERATURE - Sets temperature in OpenAI (Default: 1)
```
Comment (nponeccop) [inline .env.template]: This isn't right

**Review by nponeccop — CHANGES_REQUESTED** [review]

File: .env.template (hunk 2 above)
Comment (riensen) [reply .env.template]: fixed

**Review by nponeccop — APPROVED** [review]

**Comment by Pwuts** [comment]
This is just making it too easy for the spammers...

**Comment by riensen** [comment]
> This is just making it too easy for the spammers...

I understand your concerns about the potential misuse by spammers. However, I would like to present some points on this:

1. **Important Use Case:** The capability to write emails is crucial for the development of sophisticated applications like virtual assistants or EntrepreneurGPT. By excluding this feature, we limit the potential of Auto-GPT and hinder its progress towards becoming a comprehensive tool for various use cases.
2. **Spammers Already Utilize GPT:** There is evidence to suggest that spammers are already using GPT-like models to enhance their strategies, as seen in this tweet: [twitter.com/AKASpencerScott/status/1638996893576601600](https://twitter.com/AKASpencerScott/status/1638996893576601600). Therefore, integrating this feature in Auto-GPT does not necessarily enable spammers. It's important to recognize that GPT, and not Auto-GPT, is the enabler in this context.
3. **Adaptation to Transformative Technologies:** It's worth noting that many groundbreaking technologies, such as drones, have initially been misused for illegal activities. Over time, however, society and regulatory bodies have adapted to minimize these negative uses while maximizing the benefits.

In conclusion, I believe the potential advantages of adding this email feature into Auto-GPT far outweigh the risks.

File: .env.template [hunk 3]
```diff
@@ -172,3 +172,12 @@ TW_CONSUMER_KEY=
 TW_CONSUMER_SECRET=
 TW_ACCESS_TOKEN=
 TW_ACCESS_TOKEN_SECRET=
+
+################################################################################
+### EMAIL (SMTP)
+################################################################################
+
```
Comment (Anasomer7) [inline .env.template:178]: Add an USE_EMAIL variable that can be set to True or False
`USE_EMAIL=False`
and initialize `self.use_email' in config.py

File: README.md [hunk 4]
```diff
@@ -462,6 +462,17 @@ HUGGINGFACE_API_TOKEN="YOUR_HUGGINGFACE_API_TOKEN"
 sudo Xvfb :10 -ac -screen 0 1024x768x24 & DISPLAY=:10 <YOUR_CLIENT>
 ```
 
+## EMail
+
+Set your sender email address and SMTP password in your `.env`. For GMail use the [App Password](https://myaccount.google.com/apppasswords) and when not using GMail adjust SMTP host and port.
+
+```bash
+EMAIL_ADDRESS=<YOUR_EMAIL_ADDRESS>
+EMAIL_ADDRESS=<YOUR_EMAIL_ADDRESS>
```
Comment (Anasomer7) [inline README.md]: correct the typo

File: autogpt/app.py [hunk 5]
```diff
@@ -192,6 +193,11 @@ def execute_command(command_name: str, arguments):
             return generate_image(arguments["prompt"])
         elif command_name == "send_tweet":
             return send_tweet(arguments["text"])
+        elif command_name == "send_email":
+            return send_email(
```
Comment (Anasomer7) [inline autogpt/app.py:222]: inside the elif command_name=='send_email":
add  `if use_email:
                return send_email()
          else:
               return("email not set up"
`

File: autogpt/commands/email.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import os
+from dotenv import load_dotenv
+import smtplib
+from email.message import EmailMessage
+
+load_dotenv()
```
Comment (Anasomer7) [inline autogpt/commands/email.py]: better initialize .env variables in config.py

**Review by Anasomer7 — CHANGES_REQUESTED** [review]

File: autogpt/commands/email.py (hunk 6 above)
Comment (Anasomer7) [reply autogpt/commands/email.py]: i created a pull request 2 days ago to add email feature . iam suggesting all this changes from that.

File: README.md (hunk 4 above)
Comment (riensen) [reply README.md]: thx done

File: autogpt/commands/email.py (hunk 6 above)
Comment (riensen) [reply autogpt/commands/email.py]: thx done

Comment (riensen) [reply autogpt/commands/email.py]: Let's get either merged, I briefly looked at yours, and you have many superfluous files in the files changed. Plus some other stuff. Would be good to have them more aligned.

File: .env.template (hunk 3 above)
Comment (riensen) [reply .env.template:178]: USE_EMAIL=False is unnecessary, because if the send address and pwd is not set, then obviously no emails can be sent. It's the same things. Also it would break convention with the how the other commands handle it.

File: autogpt/app.py (hunk 5 above)
Comment (riensen) [reply autogpt/app.py:222]: no this is not necessary and breaks convention on how the other commands work. See also my comment above.

**Comment by riensen** [comment]
@nponeccop please reapprove after I made some minor changes.

**Comment by riensen** [comment]
@nponeccop should be good now

**Comment by Pwuts** [comment]
@riensen sorry for the delay in merging. We appreciate the effort and are currently working to ship plugin support above all else (#757, expected to drop end of this week), to better accommodate for contributions like this, because we are getting a _lot_.

If you need help turning it into a plugin, please ping [at]BillSchumacher on the discord server. :)

**Comment by riensen** [comment]
@Pwuts I understand that plugin support is a priority, and I appreciate your help in accommodating contributions like mine.

In the end it is a decision between implementing the email sending feature as a plugin or built-in. A point can be made that sending emails is a fundamental functionality that could potentially benefit a wide range of users. Moreover, since this feature relies solely on core libraries, it doesn't introduce any additional dependencies (see also files changed).

In any case I am open to exploring the plugin system and will get in touch with @BillSchumacher on the Discord server.

**Comment by Pwuts** [comment]
Thanks for understanding! In our current vision, e-mail is not essential/fundamental functionality, even though it isn't a very code-heavy feature.

Closing the PR for now, hoping to see it return as a plugin. :)

**Comment by Pwuts** [comment]
Plugin support just merged to `master`. (Note: subject to change until `v0.3.0` release)

**Comment by riensen** [comment]
I am excited to share that I have successfully developed the email writing feature as a plugin for Auto-GPT. You can find the plugin on GitHub at the following link:

https://github.com/riensen/Auto-GPT-Email-Plugin

The plugin is fully functional and offers capabilities for both checking and sending emails. I encourage you to try it out and provide any feedback you may have.

I'm looking forward to hearing your thoughts and continuing to contribute to the Auto-GPT project.
@BillSchumacher @Pwuts
